receiver/dcgmreceiver/scraper_gpu_test.go

```diff
@@ -48,20 +61,60 @@ func TestScrapeWithGpuPresent(t *testing.T) {
 	err := scraper.start(context.Background(), componenttest.NewNopHost())
 	require.NoError(t, err)
 
-	metrics, err := scraper.scrape(context.Background())
+	metrics, err := collectScraperResult(t, context.Background(), scraper)
+	assert.NoError(t, err)
+
+	assert.NoError(t, scraper.stop(context.Background()))
+
+	validateScraperResult(t, metrics)
+}
+
+func TestScrapeCollectionInterval(t *testing.T) {
+	var settings receiver.CreateSettings
+	settings.Logger = zaptest.NewLogger(t)
+
+	var fetchCount int
+
+	realDcgmGetValuesSince := dcgmGetValuesSince
+	defer func() { dcgmGetValuesSince = realDcgmGetValuesSince }()
+	dcgmGetValuesSince = func(g dcgm.GroupHandle, f dcgm.FieldHandle, t time.Time) ([]dcgm.FieldValue_v2, time.Time, error) {
+		fetchCount++
+		return realDcgmGetValuesSince(g, f, t)
+	}
+
+	scraper := newDcgmScraper(createDefaultConfig().(*Config), settings)
+	require.NotNil(t, scraper)
+
+	err := scraper.start(context.Background(), componenttest.NewNopHost())
+	require.NoError(t, err)
+
+	// We expect to scrape every maxKeepSamples * scrapePollingInterval / 2.
+	// Wait long enough that we expect three scrapes.
+	const sleepTime = 3.5 * maxKeepSamples * scrapePollingInterval / 2
+
+	time.Sleep(sleepTime)
+
+	metrics, err := collectScraperResult(t, context.Background(), scraper)
 	assert.NoError(t, err)
 
-	require.NotNil(t, scraper.client)
-	require.NotEmpty(t, scraper.client.devicesModelName)
-	expectedMetrics := loadExpectedScraperMetrics(t, scraper.client.getDeviceModelName(0))
-	validateScraperResult(t, metrics, expectedMetrics)
+	assert.NoError(t, scraper.stop(context.Background()))
+
+	// We should have seen 1 initial scrape + 3 timed scrapes + 2 scrapes triggered by `collectScraperResult`.
```

Comment:
```suggestion
	// We should have seen 1 initial scrape + 3 timed scrapes + up to 2 scrapes triggered by `collectScraperResult`.
```
Only cumulative metrics trigger 2 scrapes, right?

Reply:
`collectScraperResult` is a test helper at the top of this file. It always scrapes until it gets a metric value (which in this particular test should always be present), and then scrapes one more time (to make sure we have values for cumulative metrics). Those are the 2 scrapes I'm counting here. `collectScraperResult` doesn't have any way of knowing if there's a missing cumulative metric or not, so it doesn't implement "up to", it just always scrapes twice.